includes/Editor.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Class editor.
+ *
+ * Editor class, with output buffer.
+ *
+ * @package   Google\Web_Stories\Traits
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories_Dependencies\AmpProject\Dom\Document;
+
+/**
+ * Class Editor
+ *
+ * @package Google\Web_Stories
+ */
+class Editor {
+
+	/**
+	 * Experiments instance.
+	 *
+	 * @var Experiments Experiments instance.
+	 */
+	private $experiments;
+
+	/**
+	 * Editor constructor.
+	 *
+	 * @param Experiments $experiments Experiments instance.
+	 */
+	public function __construct( Experiments $experiments ) {
+		$this->experiments = $experiments;
+	}
+	/**
+	 * Init
+	 *
+	 * @return void
+	 */
+	public function init() {
+		if ( ! $this->experiments->is_experiment_enabled( 'videoOptimization' ) ) {
+			return;
+		}
```

Comment:
Need to check user meta here as well. No need for doing that if user has opted out.

Reply:
Fixed.